include/mbedtls/arc4.h

```diff
@@ -99,13 +99,17 @@ int mbedtls_arc4_crypt( mbedtls_arc4_context *ctx, size_t length, const unsigned
 extern "C" {
 #endif
 
+#if defined(MBEDTLS_SELF_TEST)
+
 /**
  * \brief          Checkup routine
  *
  * \return         0 if successful, or 1 if the test failed
  */
 int mbedtls_arc4_self_test( int verbose );
 
+#endif /*MBEDTLS_SELF_TEST*/
```

Comment:
I thought about it, but then it is risky if some will add new public APIs which should not be replaced by alternative implementation, and miss that it's surrounded by the `extern "C" {}` block